packages/block-library/src/comments-title/edit.js

```diff
@@ -0,0 +1,148 @@
+import apiFetch from '@wordpress/api-fetch';
+import { addQueryArgs } from '@wordpress/url';
+
+function Edit( {
+	attributes: {
+		textAlign,
+		singleCommentLabel,
+		multipleCommentsLabel,
+		showSingleCommentLabel,
+		showPostTitle,
+	},
+	setAttributes,
+	context: { postType, postId },
+} ) {
+	const [ commentsCount, setCommentsCount ] = useState();
+
+	const [ rawTitle ] = useEntityProp( 'postType', postType, 'title', postId );
+
+	const blockProps = useBlockProps( {
+		className: classnames( {
+			[ `has-text-align-${ textAlign }` ]: textAlign,
+		} ),
+	} );
+
+	useEffect( () => {
+		if ( ! postId ) {
+			setCommentsCount( 2 );
+		}
+		const currentPostId = postId;
+		apiFetch( {
+			path: addQueryArgs( '/wp/v2/comments', {
+				post: postId,
+				_fields: 'id',
+			} ),
+			method: 'HEAD',
+			parse: false,
+		} ).then( ( res ) => {
+			// Stale requests will have the `currentPostId` of an older closure.
+			if ( currentPostId === postId ) {
+				setCommentsCount( parseInt( res.headers.get( 'X-WP-Total' ) ) );
+			}
+		} );
+	}, [ postId ] );
+
+	const blockControls = (
+		<BlockControls group="block">
+			<AlignmentControl
+				value={ textAlign }
+				onChange={ ( newAlign ) =>
+					setAttributes( { textAlign: newAlign } )
+				}
+			/>
+		</BlockControls>
+	);
+
+	const inspectorControls = (
+		<InspectorControls>
+			<PanelBody title={ __( 'Settings' ) }>
+				<ToggleControl
+					label={ __( 'Show post title' ) }
+					checked={ showPostTitle }
+					onChange={ ( value ) =>
+						setAttributes( { showPostTitle: value } )
+					}
+				/>
+				{ ! postId && (
+					<ToggleControl
+						label={ __( 'Show single comment' ) }
+						help={ __(
+							'Toggles between single comments and multiple comments reply sentence'
+						) }
+						checked={ showSingleCommentLabel }
+						onChange={ ( value ) =>
+							setAttributes( { showSingleCommentLabel: value } )
+						}
+					/>
+				) }
+			</PanelBody>
+		</InspectorControls>
+	);
+
+	const postTitle = postId ? rawTitle : 'Post Title';
+
+	return (
+		<>
+			{ blockControls }
+			{ inspectorControls }
+			<h3 { ...blockProps }>
+				{ showSingleCommentLabel || commentsCount === 1 ? (
+					<>
+						<PlainText
+							__experimentalVersion={ 2 }
+							tagName="span"
+							aria-label={ __( 'One response to ' ) }
+							placeholder={ __( 'One response to ' ) }
+							value={ singleCommentLabel }
+							onChange={ ( newLabel ) =>
+								setAttributes( {
+									singleCommentLabel: newLabel,
+								} )
+							}
+						/>
+						{ showPostTitle ? postTitle : null }
+					</>
+				) : (
+					<>
+						{ commentsCount }
+						<PlainText
+							__experimentalVersion={ 2 }
+							tagName="span"
+							aria-label={ __( ' Responses to ' ) }
+							placeholder={ __( ' Responses to ' ) }
+							value={ multipleCommentsLabel }
+							onChange={ ( newLabel ) =>
+								setAttributes( {
+									multipleCommentsLabel: newLabel,
+								} )
+							}
+						/>
```

Comment:
Why are we using `<PlainText /> here instead of just rendering a plain string?

I see that it is causing weird CSS issues in the editor (the `color` and `opacity` are inherited from some unrelated classes for example):

https://user-images.githubusercontent.com/5417266/163897688-77f82090-bd18-4182-afb6-446a8a1066c2.mov

Reply:
<PlainText /> is a placeholder where you can edit the content. It is used for example in pagination links, where you can edit the text. It seems that the opacity is a way to tell the user that text is editable.

Reply:
Hmm, does that mean the user should be able to edit this in the Site Editor? I have tried but can't seem to...

Reply:
Yep! That's the idea!

https://www.loom.com/share/0df30b861639431daa65d1ca23726d25

Reply:
Hmm, seems great. But for the life of me, I can't seem to edit the Comment Title -- neither in the Site Editor nor the Post Editor... 😕

Reply:
@DAreRodz helped me figure this out: These are actual placeholders (like the greyed-out value you see in a form input before you enter anything), so you really just click anywhere on them and start typing.

The important thing is that clicking on them _won't_ set a cursor, nor is it possible to highlight part of it (e.g. by double-clicking) -- because they're placeholders 😅 

Here's the screen recording that David shared:

https://user-images.githubusercontent.com/96308/164067622-846468ec-408a-4044-837d-1c6ca4a1ca8a.mov